src/main/java/bio/terra/common/tracing/OkHttpClientTracingInterceptor.java

```diff
@@ -0,0 +1,56 @@
+package bio.terra.common.tracing;
+
+import io.opencensus.contrib.http.HttpClientHandler;
+import io.opencensus.contrib.http.HttpRequestContext;
+import io.opencensus.trace.Tracer;
+import io.opencensus.trace.Tracing;
+import io.opencensus.trace.propagation.TextFormat.Setter;
+import java.io.IOException;
+import okhttp3.Interceptor;
+import okhttp3.Request;
+import okhttp3.Response;
+
+public class OkHttpClientTracingInterceptor implements Interceptor {
+  private Tracer tracer;
+  // This looks a little odd, but OkHttp requests are immutable so the carrier type must be a
+  // request builder, not a Request.
+  private HttpClientHandler<Request, Response, Request.Builder> handler;
+  private static final Setter<Request.Builder> SETTER =
+      new Setter<Request.Builder>() {
+        @Override
+        public void put(Request.Builder carrier, String key, String value) {
+          carrier.header(key, value);
+        }
+      };
+
+  public OkHttpClientTracingInterceptor(Tracer tracer) {
+    this.tracer = tracer;
+    this.handler =
+        new HttpClientHandler<>(
+            tracer,
+            new OkHttpTracingExtractor(),
+            Tracing.getPropagationComponent().getB3Format(),
+            SETTER);
+  }
+
+  @Override
+  public Response intercept(Chain chain) throws IOException {
+    Request originalRequest = chain.request();
+    Request.Builder newRequestBuilder = originalRequest.newBuilder();
+
+    HttpRequestContext requestContext =
+        handler.handleStart(tracer.getCurrentSpan(), newRequestBuilder, originalRequest);
+    Request newRequest = newRequestBuilder.build();
+    Response response = null;
+    try {
+      response = chain.proceed(newRequest);
+    } catch (IOException e) {
+      // Error during call
+      handler.handleEnd(requestContext, newRequest, null, e);
+      throw e;
```

Comment:
Where will this exception wind up? Should you add a message?

Reply:
Interceptors wrap the actual HTTP request, so throwing here means the exception goes up to the previous interceptor in the chain (and eventually to the client library that made the actual call). 
Since this interceptor is only handling tracing and this exception is coming from the call, I think it's better to pass the exception without modifying it here.